### DISCUSSION TITLE
How can i add custom Snippets

### DISCUSSION THREAD

**Discussion by Naimul007A** [NvChad/NvChad#2677] [Q&A] [answered]
i want to add custom Snippets.How can I do this in NvChad ?

**Comment by carlisia** [comment, marked as answer]
Hey there, I just configured this for myself. I might be missing something, but here's what I know, hope it helps.

First you need to decide what snippet formats your snippets will be in [lua|snipmate|vscode]. This section of the luasnip doc might be helpful: [[LuaSnip/DOC.md at master · L3MON4D3/LuaSnip](https://github.com/L3MON4D3/LuaSnip/blob/master/DOC.md#loaders)](https://github.com/L3MON4D3/LuaSnip/blob/master/DOC.md#loaders).

Basically, lua snippets go in a `.lua` file, and vscode in a `.json` file, one per language, with a `package.json` pointing to each of the files. Ex:
```
{
  "name": "snippets",
  "contributes": {
    "snippets": [
      {"language": "markdown" , "path": "./markdown.json"}
    ]
  }
}
```

Assuming you have an `options.lua` file, add this in there:
```
require("luasnip.loaders.from_vscode").lazy_load {
  paths = vim.g.vscode_snippets_path or vim.fn.stdpath "config" .. "/snippets",
}
```

The above is for vscode. 


The nvchad doc is here: [[NvChad Snippets](https://nvchad.com/docs/config/snippets/#globals)](https://nvchad.com/docs/config/snippets/#globals)
